File: 2352-Equal-Row-and-Column-Pairs/2352-Equal-Row-and-Column-Pairs.java
class Solution {
    boolean check(int row, int col, int[][] grid) {
        for (int i = 0; i < grid.length; i++) {
            if (grid[row][i] != grid[i][col]) {
                return false;
            }
        }
        return true;
    }
    public int equalPairs(int[][] grid) {
        int count = 0;
        int n = grid.length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][0] == grid[0][j] && check(i, j, grid)) {
                        count++;
                }
            }
        }
        return count;
        
    }
}

File: 2352-Equal-Row-and-Column-Pairs/2352-Equal-Row-and-Column-Pairs.go
func check(row, col int, grid [][]int) bool {
	for i := 0; i < len(grid); i++ {
		if grid[row][i] != grid[i][col] {
			return false
		}
	}
	return true
}

func equalPairs(grid [][]int) int {
	count := 0
	n := len(grid)

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			if grid[i][0] == grid[0][j] && check(i, j, grid) {
				count++
			}
		}
	}

	return count
}
